platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -382,50 +464,52 @@ class HangingProtocolService {
   }
 
   // Match images given a list of Studies and a Viewport's image matching reqs
-  _matchImages(displaySet) {
-    console.log('ProtocolEngine::matchImages');
+  _matchImages(displaySetRules) {
 
     // TODO: matching is applied on study and series level, instance
     // level matching needs to be added in future
 
     // Todo: handle fusion viewports by not taking the first displaySet rule for the viewport
-    const { studyMatchingRules, seriesMatchingRules } = displaySet;
+    const { studyMatchingRules = [], seriesMatchingRules, findAll = false } = displaySetRules;
 
     const matchingScores = [];
     let highestStudyMatchingScore = 0;
     let highestSeriesMatchingScore = 0;
 
+    console.log('ProtocolEngine::matchImages', studyMatchingRules, seriesMatchingRules);
     this.studies.forEach(study => {
+      // TODO - only pass in display sets matching the top level study being tested
```

Comment:
This is an easy todo to fix, `this.displaySets.filter(ds => ds.StudyInstanceUID..........)